build-logic/src/main/kotlin/dokkabuild/DokkaBuildProperties.kt

```diff
@@ -48,11 +49,46 @@ abstract class DokkaBuildProperties @Inject constructor(
     val kotlinLanguageLevel: Provider<KotlinVersion> =
         dokkaProperty("kotlinLanguageLevel", KotlinVersion::fromVersion)
 
+    val integrationTestParallelism: Provider<Int> =
+        dokkaProperty("integration_test.parallelism", String::toInt)
+            // fallback & warn if the old property is used
+            .orElse(
+                providers.gradleProperty("dokka_integration_test_parallelism")
+                    .map {
+                        logger.warn("dokka_integration_test_parallelism is deprecated - use org.jetbrains.dokka.integration_test.parallelism instead")
+                        it.toInt()
+                    }
+            )
+
+    val integrationTestExhaustive: Provider<Boolean> =
+        dokkaProperty("integration_test.exhaustive", String::toBoolean)
+            // fallback & warn if the old property is used
+            .orElse(
+                providers.gradleProperty("dokka_integration_test_is_exhaustive")
+                    .map {
+                        logger.warn("dokka_integration_test_is_exhaustive is deprecated - use org.jetbrains.dokka.integration_test.exhaustive instead")
+                        it.toBoolean()
+                    }
+            )
+            .orElse(
+                providers.environmentVariable("DOKKA_INTEGRATION_TEST_IS_EXHAUSTIVE")
+                    .map {
+                        logger.warn("DOKKA_INTEGRATION_TEST_IS_EXHAUSTIVE is deprecated - use ORG_GRADLE_PROJECT_org.jetbrains.dokka.integration_test.exhaustive instead")
+                        it.toBoolean()
+                    }
+            )
+
+    val tryK2: Provider<Boolean> =
```

Comment:
BTW: while we are here, if this is used only in integration tests, may be it's better to rename it to `integration_test.useK2` instead of using the same flag which is used to run `dokka` with K2 - now it feels a bit confusing IMO

Reply:
Good idea. Just to be clear the full path will be `org.jetbrains.dokka.integration_test.useK2`

Reply:
Note: TC config should be updated to use this new property - after merge to master only